pkg/framework/host.go

```diff
@@ -144,46 +144,51 @@ func (h *Host) DeleteGuestCluster() error {
 	return h.WaitForPodLog("giantswarm", logEntry, operatorPodName)
 }
 
-func (h *Host) InstallAwsOperator(values string) error {
-	awsOperatorChartValuesEnv := os.ExpandEnv(values)
+func (h *Host) InstallStableOperator(name, cr, values string) error {
+	err := h.installOperator(name, cr, values, ":stable")
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	return nil
+}
+
+func (h *Host) InstallBranchOperator(name, cr, values string) error {
+	err := h.installOperator(name, cr, values, "@1.0.0-${CIRCLE_SHA1}")
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	return nil
+}
 
-	tmpfile, err := ioutil.TempFile("", "aws-operator-values")
+func (h *Host) installOperator(name, cr, values, version string) error {
+	chartValuesEnv := os.ExpandEnv(values)
+
+	tmpfile, err := ioutil.TempFile("", name+"-values")
 	if err != nil {
 		return microerror.Mask(err)
 	}
 	defer os.Remove(tmpfile.Name())
 
-	if _, err := tmpfile.Write([]byte(awsOperatorChartValuesEnv)); err != nil {
+	if _, err := tmpfile.Write([]byte(chartValuesEnv)); err != nil {
 		return microerror.Mask(err)
 	}
 
+	cmd := fmt.Sprintf("registry install quay.io/giantswarm/%[1]s-chart%[2]s -- -n %[1]s --values %[3]s", name, version, tmpfile.Name())
```

Comment:
How does the formatting with the square brackets work? Never saw this before.

Reply:
I also had to educate myself: https://golang.org/pkg/fmt/ -> search for `Explicit argument indexes`.

Reply:
Exactly, here, given that the name is used twice in the format string, the goal is not having to repeat values, see https://play.golang.org/p/IfqFTpiV5eh for example in contrast to https://play.golang.org/p/gnzXPMw2J5F